services/wallet/activity/activity.go

```diff
@@ -96,7 +96,8 @@ func newActivityEntryWithTransaction(pending bool, transaction *transfer.Transac
 		timestamp:      timestamp,
 		activityType:   activityType,
 		activityStatus: activityStatus,
-		tokenType:      AssetTT,
+		// TODO: update tokenType
+		tokenType: Native,
```

Comment:
Maybe this should be two fields "inToken" and "outToken"? Since swap will involve two different ones. Simple transfers should just fill in the appropriate one (in for Receive, out for Send). WDYT?
Same as with the chainID, the bridge operation will involve an "inChainID" and an "outChainID"

Reply:
Actually wondering if we need to have the Native-Erc20 distinction instead of calling them something like "Fungible" with the "Native" option simply being represented by symbol "ETH" or an empty contract address. With this approach, we would have:
-Fungible (ETH+ERC20) -> Identified by contract address or symbol name. Transaction uses `Amount` field
-Non-Fungible (ERC721) -> Identified by contract address and tokenID. Doesn't use `Amount` field (only 1 token per tokenID in existence)
-Semi-Fungible (ERC1155) -> Identified by contract address and tokenID. Uses `Amount` field

With this new definitio, the `tokenType` is unique for each activty entry (instead of the current approach, where you can swap `Native` and `erc20` tokens).

Just throwing ideas around!

Reply:
> Maybe this should be two fields "inToken" and "outToken"?
Sounds great. However, won't `sentToken` and `receivedToken` bring more clarity?

> Actually wondering if we need to have the Native-Erc20 ...

I think I prefer the current explicit approach; the others are simpler but hide complexity by requiring to have the understanding you mentioned. Not sure if there are other benefits to the fungible/non-semi-fungible approach